### PR TITLE
Warn when getBestCameraToTarget returns 0, 0, 0

### PR DESCRIPTION
**Pull request by spacey-sooty**
Resolves https://github.com/PhotonVision/photonvision/issues/915

### CODE REVIEW

File: photon-core/src/main/java/org/photonvision/vision/target/TrackedTarget.java [hunk 1]
```diff
@@ -364,6 +366,9 @@ public boolean hasSubContours() {
     }
 
     public Transform3d getBestCameraToTarget3d() {
+        if (m_bestCameraToTarget3d == new Transform3d()) {
+            DriverStation.reportWarning("Camera has not been calibrated.", false);
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/vision/target/TrackedTarget.java:370]: This warning might be spammed. Also, this transform won't be set if the pipeline isn't in 3d mode -- you don't know which reason it is. 

This functionality in c++ too would be good

Comment (spacey-sooty) [reply photon-core/src/main/java/org/photonvision/vision/target/TrackedTarget.java:370]: > Also, this transform won't be set if the pipeline isn't in 3d mode
Should I add a note about that in the warning?

**Review by mcm001 — APPROVED** [review]
🛳️ it? Fix it in prod?

File: photon-core/src/main/java/org/photonvision/vision/target/TrackedTarget.java [hunk 2]
```diff
@@ -364,6 +365,9 @@ public boolean hasSubContours() {
     }
 
     public Transform3d getBestCameraToTarget3d() {
+        if (m_bestCameraToTarget3d == new Transform3d()) {
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/vision/target/TrackedTarget.java:368]: This equality check checks the reference, and not the equals function

File: photon-targeting/src/main/native/include/photon/targeting/PhotonTrackedTarget.h [hunk 3]
```diff
@@ -138,7 +139,12 @@ class PhotonTrackedTarget {
    * reprojection error is the ambiguity, which is between 0 and 1.
    * @return The pose of the target relative to the robot.
    */
-  frc::Transform3d GetBestCameraToTarget() const { return bestCameraToTarget; }
+  frc::Transform3d GetBestCameraToTarget() const {
+    if (bestCameraToTarget == frc::Transform3d()) {
```
Comment (mcm001) [inline photon-targeting/src/main/native/include/photon/targeting/PhotonTrackedTarget.h:143]: Transform3d does override operator== yeah?

**Review by mcm001 — CHANGES_REQUESTED** [review]

**Review by mcm001 — APPROVED** [review]

**Pull request merged**